dev/PushNotifications/GetRawNotificationEventArgs.h

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+#include "PushNotificationReceivedEventArgs.h"
+#include <winrt/Windows.ApplicationModel.Core.h>
+#include "externs.h"
+
+static PCWSTR c_pushContractId = L"Windows.Push";
+
+using namespace winrt::Windows::ApplicationModel::Core;
+namespace winrt::Microsoft::Windows::PushNotifications
+{
+    static winrt::Windows::Foundation::IInspectable Deserialize(winrt::Windows::Foundation::Uri const&)
+    {
+        auto appProperties = CoreApplication::Properties();
+        if (auto foundActivatedEventArgs = appProperties.TryLookup(ACTIVATED_EVENT_ARGS_KEY))
+        {
+            return foundActivatedEventArgs.as<PushNotificationReceivedEventArgs>();
+        }
+
+        THROW_HR_IF_NULL_MSG(E_UNEXPECTED, g_waitHandleForArgs, "PushNotificationManager::RegisterActivator has not been called.");
+
+        if (WaitForSingleObject(g_waitHandleForArgs.get(), 1000) == WAIT_OBJECT_0)
+        {
+            appProperties = CoreApplication::Properties();
+            if (auto foundActivatedEventArgs = appProperties.TryLookup(ACTIVATED_EVENT_ARGS_KEY))
```

Comment:
Did you confirm whether these API calls have thread safety built into them? What if I am writing into this property set at the same time I am reading out of it?

Reply:
Looking with @oldnewthing offline.

Reply:
The COM static store is thread safe itself.